x-pack/plugins/security_solution/server/endpoint/routes/actions/service.ts

```diff
@@ -9,6 +9,60 @@ import { Logger } from 'kibana/server';
 import type { estypes } from '@elastic/elasticsearch';
 import { AGENT_ACTIONS_INDEX, AGENT_ACTIONS_RESULTS_INDEX } from '../../../../../fleet/common';
 import { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { ActivityLog } from '../../../../common/endpoint/types';
+
+const getShouldClauses = ({
+  indices,
+  elasticAgentId,
+}: {
+  indices: string[];
+  elasticAgentId: string;
+}) => {
+  return indices.map((_index) => {
+    const clause = {
+      bool: {
+        must: [
+          {
+            match: {
+              _index,
+            },
+          },
+          {
+            match: {
+              [_index === AGENT_ACTIONS_INDEX ? 'agents' : 'agent_id']: elasticAgentId,
+            },
+          },
+        ],
+      },
+    };
+    // only `endpoint` actions in the result
+    if (_index === AGENT_ACTIONS_INDEX) {
+      clause.bool.must.push(
+        {
+          match: {
+            input_type: 'endpoint',
+          },
+        },
+        {
+          match: {
+            type: 'INPUT_ACTION',
+          },
+        }
+      );
+    } else {
+      // only when `action_data` field is present (not indexed)
+      // `started_at` appears along with `action_data`
+      clause.bool.must.push({
+        // @ts-expect-error
+        // this is a valid clause, yet not in TS
+        exists: {
+          field: 'started_at',
+        },
```

Comment:
We should not be relying on this behavior. I don't think this is a stable query to use for the results

Reply:
quite agree. I'll remove this, now that we're filtering with matching `action_id`s.